castai/resource_organization_group.go

```diff
@@ -0,0 +1,229 @@
+package castai
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/samber/lo"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+)
+
+const (
+	FieldOrganizationGroupOrganizationID = "organization_id"
+	FieldOrganizationGroupName           = "name"
+	FieldOrganizationGroupDescription    = "description"
+	FieldOrganizationGroupMembers        = "members"
+	FieldOrganizationGroupMember         = "member"
+	FieldOrganizationGroupMemberID       = "id"
+	FieldOrganizationGroupMemberEmail    = "email"
```

Comment:
```suggestion
	FieldOrganizationGroupMemberEmail    = "email"
	FieldOrganizationGroupMemberKind     = "kind" # USER or SERVICE_ACCOUNT
```
Had to introduce an extra attribute to distinct whether a group member is a user or a serviceAccount as both cases are valid from the business perspective - https://api.cast.ai/v1/spec/#/RbacServiceAPI/RbacServiceAPI_CreateGroup

Reply:
@radekska I have added support for member kind